slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+import enum
+import logging
+import os
+import pathlib
+import shutil
+import time
+from pathlib import Path
+from typing import Any, Mapping, Optional, cast
+
+import httpx
+import requests
+import slay
+import truss
+from slay import definitions, utils
+from slay.utils import ConditionStatus
+from truss import truss_config
+from truss.contexts.image_builder import serving_image_builder
+from truss.remote import remote_factory, truss_remote
+from truss.remote.baseten import service as b10_service
+
+_REQUIREMENTS_FILENAME = "pip_requirements.txt"
+_MODEL_CLASS_FILENAME = "processor.py"
+_MODEL_CLASS_NAME = "ProcessorModel"
+_TRUSS_DIR = ".truss_gen"
+
+
+def _copy_python_source_files(root_dir: pathlib.Path, dest_dir: pathlib.Path) -> None:
+    """Copy all python files under root recursively, but skips generated code."""
+
+    def python_files_only(path, names):
+        return [
+            name
+            for name in names
+            if os.path.isfile(os.path.join(path, name))
+            and not name.endswith(".py")
+            or definitions.GENERATED_CODE_DIR in name
+        ]
+
+    shutil.copytree(root_dir, dest_dir, ignore=python_files_only, dirs_exist_ok=True)
+
+
+def _make_truss_config(
```

Comment:
as we discussed, I think it would be cool if we could alternatively point to an existing config.yaml file. can do this in a follow-up

Reply:
Yes, I've added a TODO. It's not that trivial, if we also extract the I/O types... so let's do this in a follow up.